### ISSUE TITLE
Sodiumoxide requires dynamically linked glibc: version `GLIBC_2.25' not found on remote

### ISSUE THREAD

**Issue by aufziehvogel** [sodiumoxide/sodiumoxide#399]
Hello,

according to my understanding current versions of sodiumoxide should automatically build a static binary with all dependencies included (since https://github.com/sodiumoxide/sodiumoxide/pull/281). However, with the following minimal example I cannot execute my locally compiled binary on my remote server.

**Cargo.toml**
```toml
[package]
name = "foo"
version = "0.1.0"
authors = ["aufziehvogel"]
edition = "2018"

[dependencies]
sodiumoxide = "0.2.5"
```

**src/main.rs**
```rust
use sodiumoxide::crypto::secretbox;

fn main() {
    let key = secretbox::gen_key();
    let nonce = secretbox::gen_nonce();
    let plaintext = b"some data";
    let ciphertext = secretbox::seal(plaintext, &nonce, &key);
    let their_plaintext = secretbox::open(&ciphertext, &nonce, &key).unwrap();

    println!("{}", std::str::from_utf8(&their_plaintext).unwrap());
}
```

I compiled with:

```bash
$ cargo build --release
```

Locally it runs fine with:

```bash
$ ./target/release/foo
some data
```

On my server however it will give me:

```bash
$ ./foo
./foo: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by ./foo)
```

Is there anything I can do to help you track this issue down?

Regards
Stefan

PS: `rust_sodium` shows exactly the same behaviour. So it's quite possible I am misunderstanding something. However, I have worked with other libraries that pull in C dependencies and that worked fine (e.g. `git2` with the features `features = ["vendored-openssl"]`).

**Comment by aufziehvogel** [comment]
Hm, sorry I sent the issue to quickly... I think it hasn't worked before. I think before I [built with MUSL](https://doc.rust-lang.org/1.9.0/book/advanced-linking.html) when I copied to my server.

**Comment by kpp** [comment]
Do `ldd ./target/release/foo` locally; `ldd ./foo` and `dpkg -S libc.so` on the server and show me the output. Looks like libc is outdated on the server.

**Comment by aufziehvogel** [comment]
Yes it is. It's a Debian system :)

I was just confused and didn't remember that in order to have libc statically compiled in Rust you need musl.

Sorry for that.

Just for the sake of completion, my ldd output:

```
~$ ldd foo 
./foo: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by ./foo)
	linux-vdso.so.1 (0x00007ffc671ee000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fe667d7b000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fe667b5e000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fe667947000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fe6675a8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fe667f7f000)
```

**Comment by kpp** [comment]
```
sudo apt-get update
sudo apt-get install libc6
```

on the server and try to run `./foo` again

**Comment by aufziehvogel** [comment]
`libc6 is already the newest version (2.24-11+deb9u4)`

As far as I remember that's just a version difference between my local system and server system.

Don't worry about it. If I remember correctly my final decision about the libc was: "Ok, I tested it with MUSL and it would work. But since it's easier I will just compile with the server when needed". So that's where my confusion came from, because I had it already working (but with musl), and because I decided against musl in the end I did not have any special configuration in my repo.
